src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,306 @@
+            case KC_WATER_TAXI_VASHON_ISLAND: return 4.50f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 3.75f;
+            case KITSAP_TRANSIT: return 2.00f;
+            case KC_METRO:
+            case SOUND_TRANSIT:
+            case EVERETT_TRANSIT:
+            case SEATTLE_STREET_CAR:
+                return 1.50f;
+            default: return defaultFare;
+        }
+    }
+
+    /**
+     * Get the ride price for a single leg. If testing, this class is being called directly so the required agency cash
+     * values are not available therefore the default test price is used instead.
+     */
+    private float getRidePrice(Ride ride, Fare.FareType fareType, Collection<FareRuleSet> fareRules) {
+        if (IS_TEST) {
+            // Testing, return default test ride price.
+            return DEFAULT_TEST_RIDE_PRICE;
+        }
+        return calculateCost(fareType, Lists.newArrayList(ride), fareRules);
+    }
+
+    /**
+     * Calculate the cost of a journey. Where free transfers are not permitted the cash price is used. If free transfers
+     * are applicable, the most expensive discount fare across all legs is added to the final cumulative price.
+     *
+     * The computed fare for Orca card users takes into account realtime trip updates where available, so that, for
+     * instance, when a leg on a long itinerary is delayed to begin after the initial two hour window has expired,
+     * the calculated fare for that trip will be two one-way fares instead of one.
+     */
+    @Override
+    public boolean populateFare(Fare fare,
+                                   Currency currency,
+                                   Fare.FareType fareType,
+                                   List<Ride> rides,
+                                   Collection<FareRuleSet> fareRules
+    ) {
+        Long freeTransferStartTime = null;
+        float cost = 0;
+        float orcaFareDiscount = 0;
+        for (Ride ride : rides) {
+            RideType rideType = classify(ride.routeData);
+            if (freeTransferStartTime == null && permitsFreeTransfers(rideType)) {
+                // The start of a free transfer must be with a transit agency that permits it!
+                freeTransferStartTime = ride.startTime;
+            }
+            float singleLegPrice = getRidePrice(ride, fareType, fareRules);
+            float discountedFare = getDiscountedFare(fareType, rideType, singleLegPrice);
+            if (hasFreeTransfers(fareType, rideType) && inFreeTransferWindow(freeTransferStartTime, ride.startTime)) {
+                // If using Orca (free transfers), the total fare should be equivalent to the
+                // most expensive leg of the journey.
+                orcaFareDiscount = Float.max(orcaFareDiscount, discountedFare);
+            } else {
+                // If free transfers are not permitted, add the cash price of this leg to the total cost.
+                cost += singleLegPrice;
```

Comment:
Should this be as follows?

```suggestion
                cost += discountedFare;
```

It seems that with WSF and Skagit Transit, they have discounted fare types that are possible to achieve without an Orca card.

Reply:
Updated implemented for review.